agent/src/main/java/com/teamscale/jacoco/agent/options/ClasspathUtils.java

```diff
@@ -0,0 +1,38 @@
+package com.teamscale.jacoco.agent.options;
+
+import org.conqat.lib.commons.filesystem.FileSystemUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** Handles parsing a .txt file with classpath pattern separated by newlines. */
+public class ClasspathUtils {
+
+	/** Replaces all txt files in the given list with the file names written in the txt file separated by new lines. */
+	public static List<File> resolveClasspathTextFiles(String key, FilePatternResolver filePatternResolver,
+													   List<String> classDirectoriesOrZips) throws AgentOptionParseException {
+		List<File> classDirectoriesOrZipFiles = new ArrayList<>();
+		for (String classDirectoryOrZip : classDirectoriesOrZips) {
+			classDirectoriesOrZipFiles.addAll(filePatternResolver.resolveToMultipleFiles(key, classDirectoryOrZip));
+		}
+		Map<Boolean, List<File>> filesByType = classDirectoriesOrZipFiles.stream()
+				.collect(Collectors.partitioningBy(fileName -> fileName.getName().endsWith(".txt")));
+		List<File> classJarOrDirFiles = new ArrayList<>(filesByType.get(false));
+		for (File txtFile : filesByType.get(true)) {
```

Comment:
extract a method `resolveClassPathEntries` or something like that for this `for` loop? It could return a list of resolved files that you add with `addAll`. I think that would make this method easier to read :-)